scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesRetrieverTest.kt

```diff
@@ -0,0 +1,1312 @@
+      listOf(
+        MavenListDependency(
+          coord = DEP_WITH_SCRAPABLE_LICENSE,
+          url = "${DATA_BINDING_POM.dropLast(3)}aar"
+        ),
+        MavenListDependency(
+          coord = DEP_WITH_NO_LICENSE,
+          url = "${PROTO_LITE_POM.dropLast(3)}jar"
+        )
+      )
+    )
+    assertThat(mavenDependencyList.mavenDependencyList.size).isEqualTo(2)
+    val dependency1 = mavenDependencyList.mavenDependencyList[0]
+    val dependency2 = mavenDependencyList.mavenDependencyList[1]
+    assertIsDependency(
+      dependency = dependency1,
+      artifactName = DEP_WITH_SCRAPABLE_LICENSE,
+      artifactVersion = DATA_BINDING_VERSION
+    )
+    assertThat(dependency1.licenseList.size).isEqualTo(1)
+    verifyLicenseHasVerifiedLinkNotSet(
+      license = dependency1.licenseList[0],
+      licenseName = "The Apache License, Version 2.0",
+      originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+    )
+    assertIsDependency(
+      dependency = dependency2,
+      artifactName = DEP_WITH_NO_LICENSE,
+      artifactVersion = PROTO_LITE_VERSION
+    )
+    assertThat(dependency2.licenseList).isEmpty()
+  }
+
+  @Test
+  fun testGenerateDepsListFromMavenInstall_emptyBazelQueryDeps_returnsEmptyList() {
+    val mavenInstallFile = tempFolder.newFile("third_party/maven_install.json")
+    writeMavenInstallJson(mavenInstallFile)
+    val mavenListDependencies = MavenDependenciesRetriever
+      .generateDependenciesListFromMavenInstall(
+        "${tempFolder.root}/third_party/maven_install.json",
+        listOf()
+      )
+    assertThat(mavenListDependencies).isEmpty()
+  }
+
+  @Test
+  fun testGenerateDepsListFromMavenInstall_nonEmptyBazelQueryDepNames_returnsCorrectList() {
+    val mavenInstallFile = tempFolder.newFile("third_party/maven_install.json")
+    writeMavenInstallJson(mavenInstallFile)
+    val mavenListDependencies = MavenDependenciesRetriever
+      .generateDependenciesListFromMavenInstall(
+        "${tempFolder.root}/third_party/maven_install.json",
+        listOf(
+          omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_SCRAPABLE_LICENSE),
+          omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_DIRECT_LINK_ONLY_LICENSE),
+        )
+      )
+    assertThat(mavenListDependencies.size).isEqualTo(2)
+    assertThat(mavenListDependencies).contains(
+      MavenListDependency(
+        coord = DEP_WITH_SCRAPABLE_LICENSE,
+        url = "${DATA_BINDING_POM.dropLast(3)}aar"
+      )
+    )
+    assertThat(mavenListDependencies).contains(
+      MavenListDependency(
+        coord = DEP_WITH_DIRECT_LINK_ONLY_LICENSE,
+        url = "${FIREBASE_ANALYTICS_POM.dropLast(3)}aar"
+      )
+    )
+    assertThat(mavenListDependencies).doesNotContain(
+      MavenListDependency(
+        coord = DEP_WITH_INVALID_LINKS,
+        url = "${IO_FABRIC_POM.dropLast(3)}aar"
+      )
+    )
+  }
+
+  private fun verifyLicenseHasScrapableVerifiedLink(
```

Comment:
Per earlier comments, is it possible to replace all of these assert/verify methods with more direct proto comparison/contains, instead?

Reply:
We definitely can do this, but then we'll have to construct the expected dependency list as it was done here - https://github.com/oppia/oppia-android/pull/3434#discussion_r671427322

Reply:
Ack. Let's keep this as-is then; it seems pretty clear.